File: Everything/ClientApp/src/models/todo/ToDoBoard.ts
export default class ToDoBoard {
    id: number = 0;
    name: string | null = null;
    createdDate: Date | null = null;
    description: string = '';
    sequence: number = 0;

    constructor(init?: Partial<ToDoBoard>) {
        Object.assign(this, init);
    }
}

File: Everything/Models/ToDoList/ToDoBoard.cs
using System;
using System.Collections.Generic;

namespace everything.Models
{
    public class ToDoBoard : IForUser
    {
        public ToDoBoard()
        {
            ToDoColumns = new List<ToDoColumn>();
        }

        public int Id { get; set; }
        public DateTime CreatedDate { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int Sequence { get; set; }

        public int UserId { get; set; }
        public virtual User User { get; set; }

        public virtual IEnumerable<ToDoColumn> ToDoColumns { get; set; }
    }
}
